crates/runtime/src/model/tool_use.rs

```diff
@@ -213,6 +217,72 @@ impl ToolUsingChat {
 
         Ok(Some(messages))
     }
+
+    async fn _chat_request(
```

Comment:
```suggestion
    async fn chat_request(
```

Why the leading underscore?

Reply:
`_chat_request` and `_chat_stream` are the recursive underlying functions that get called. They might have extra parameters than the `Chat` trait.